common/src/main/java/com/wynntils/screens/base/widgets/SearchWidget.java

```diff
@@ -14,17 +14,19 @@
 import com.wynntils.utils.render.type.HorizontalAlignment;
 import com.wynntils.utils.render.type.TextShadow;
 import com.wynntils.utils.render.type.VerticalAlignment;
+import com.wynntils.utils.type.Pair;
 import java.util.Objects;
 import java.util.function.Consumer;
 import net.minecraft.client.gui.Font;
 import net.minecraft.network.chat.Component;
 import net.minecraft.sounds.SoundEvents;
 import org.lwjgl.glfw.GLFW;
 
-public class SearchWidget extends TextInputBoxWidget {
-    private static final Component DEFAULT_TEXT =
+public abstract class SearchWidget<S> extends TextInputBoxWidget {
```

Comment:
Do you think this refactor of this class is strictly needed? The way I see it is that you can just extend SearchWidget in `ItemSearchWidget`. I am not strictly opposed to this, but it does feel like loads of unnecessary change.

Reply:
Seeing the changes in WynntilsListScreen, it made my feeling stronger about not keeping this change, if it is not needed. (I don't currently see why it would be, at the moment.)

Reply:
The thing is, `WynntilsListScreen` has a `searchWidget` attribute, and uses `reloadElementsList(String searchTerm)` in its `reloadElements(String searchterm)` **private** method.

The way I see it is that `WynntilsItemGuideScreen` and `WynntilsIngredientGuideScreen` cannot inherit from a class where `reloadElementsList(String searchTerm)` is implemented, because it makes no sense.

In the mean time, I wanted to keep the logic in the overrides of `charTyped()` and `keyPressed()` for example.

Having a generic type for the search query was the simplest and lightest way I found to refactor the less I could. The other ways I could think of were either heavier or seemed wrong to me.

I'll look into it again but I really can't guarantee I'll figure out a better and lighter way to do it.

Reply:
I can look into this more deeply tomorrow, so I can find something I can accept (or keep this).

Reply:
What do you think of having `WynntilsGuideScreen` extending `WynntilsListScreen`  and adding an empty implementation for `reloadElementsList(String searchTerm)`. You can create your own reload method with the custom filters, however you want. And you just override the method calling `reloadElementsList` since it is only called in 1 location. That should bring down the amount of changes drastically. I know it will be a bit weird inheritance wise, but I think it is weird either way.. and this way is much less code.

Reply:
That's the way I did it originally, but then I thought it was not right inheritance wise as you mentioned it.
If this is fine for you, I also think this is the best option to reduce the amount of "unnecessary" change.

Reply:
Yeah, it is okay that way. Thanks.